bin/car_envoy.sh

```diff
@@ -102,15 +103,21 @@ darwin) # https://github.com/Homebrew/homebrew-core/blob/master/Formula/envoy.rb
   ${car} --strip-components 2 -qf "ghcr.io/homebrew/core/${path}:${tag}" "${formula}/${patch_version}/bin/envoy"
   ;;
 linux)
-  files="usr/local/bin/envoy"
+  envoy_binary="usr/local/bin/envoy"
+  files="$envoy_binary"
   if [ "${debug:-}" = '1' ]; then
     reference=envoyproxy/envoy-debug:$(echo "${version}" | sed 's/_debug//g')
     files="$files usr/local/bin/envoy.dwp"
   else
     reference=envoyproxy/envoy:${version}
   fi
 
-  ${car} --created-by-pattern 'ADD linux' --strip-components 2 -qf "${reference}" ${files}
+  # List all files and layers. Look for $envoy_binary. Capture last 5 lines before $envoy_binary and find
+  # the last "CreatedBy" phase. From the line containing "CreatedBy" get the command used to add the layer.
+  # That command will be used as $pattern for extract command.
+  pattern=$(${car_base} -t -vv --reference "${reference}" | grep -B 5 "${envoy_binary}" | grep "CreatedBy" | tail -1 | awk '{print $2 " " $3}') 
```

Comment:
ex this scans the all layers, especially as it doesn't use `-q` it scans all layers, not just the ones that include the file desires.